Sources/Build/BuildPlan.swift

```diff
@@ -606,6 +608,37 @@ public final class SwiftTargetBuildDescription {
         let path = derivedSources.root.appending(subpath)
         try fs.writeIfChanged(path: path, bytes: stream.bytes)
     }
+    
+    /// Generate an inaccessible resource bundle accessor, if appropriate.
+    ///
+    /// This addresses SR-13084 by creating an unavailable accessor which informs users of the underlying issue.
+    private func generateResourceUnavailableAccessor() throws {
+        // If a bundle has been generated then we shouldn't be here.
+        guard self.bundlePath == nil else { return }
+
+        let message = "target '\(target.name)' is either missing or is solely containing invalid resource references"
+        let stream = BufferedOutputByteStream()
+        stream <<< """
+        import class Foundation.Bundle
+
+        extension Foundation.Bundle {
+            @available(*, unavailable, message: "\(message)")
+            static var module: Bundle = {
+                fatalError("no resource bundle exists for this module")
+            }()
+        }
+        """
+
+        let subpath = RelativePath("resource_bundle_accessor.swift")
+
+        // Add the file to the dervied sources.
+        derivedSources.relativePaths.append(subpath)
+
+        // Write this file out.
+        // FIXME: We should generate this file during the actual build.
+        let path = derivedSources.root.appending(subpath)
+        try fs.writeIfChanged(path: path, bytes: stream.bytes)
+    }
```

Comment:
I think it would be good if we could consolidate this method with the existing one. I would think we could have three methods:
- one that does the boilerplate of writing the accessor and takes a string as parameter
- one method each, returning the desired content of the generated file

Reply:
Hopefully I've addressed what you wanted 👍  I did have a function which assumed the path as `resource_bundle_accessor.swift` instead of passing it through but it made the function feel really locked down and specific when in reality it _could_ be reused in the future -- so for that reason I've passed the path through, hope that's still alright :)